hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -300,6 +300,8 @@ public void initialize(URI name, Configuration originalConf)
     LOG.debug("Initializing S3AFileSystem for {}", bucket);
     // clone the configuration into one with propagated bucket options
     Configuration conf = propagateBucketOptions(originalConf, bucket);
+    // Initialize any custom signers
+    initCustomSigners(conf);
```

Comment:
I guess this is the first thing which needs to be done, before anything else even thinks about talking to AWS services. If so, this is probably the right place. But I would put it after `patchSecurityCredentialProviders`

Reply:
Fixed. Have moved this to just before bindAWSClient